server/src/test/java/io/crate/integrationtests/InsertIntoIntegrationTest.java

```diff
@@ -1846,4 +1846,40 @@ public void test_generated_sub_column_provided_null_is_validated() {
         execute("insert into t (id, obj) (select 1, {\"a\" = {\"b\" = null}} from sys.cluster)");
         assertThat(response.rowCount()).isEqualTo(0L);
     }
+
+    @Test
+    public void test_generated_column_can_refer_to_a_non_string_partitioned_by_column() {
+        execute("""
+            create table t(a int, parted int, gen_from_parted int as parted +1) partitioned by (parted);
+            """
+        );
+
+        // insert from values
+        execute("insert into t (a, parted) values (1, 2)");
+        refresh();
+        execute("select a, parted, gen_from_parted from t");
+        assertThat(response).hasRows("1| 2| 3");
+
+        // insert from sub-query
+        execute("insert into t (a, parted) select 2, 3");
+        assertThat(response.rowCount()).isEqualTo(1L);
+        refresh();
+        execute("select a, parted, gen_from_parted from t order by a");
+        assertThat(response).hasRows("1| 2| 3", "2| 3| 4");
+    }
+
+    @Test
+    public void test_non_string_partitioned_by_column_with_check_contraint() {
+        execute("""
+             CREATE TABLE t (
+                 a INT,
+                 b INT CONSTRAINT check_1 CHECK (b>10)
+             ) PARTITIONED BY (b)
+             """
+        );
+
+        // Partitioned column value 12 used to be provided as "12" internally and record was rejected with casting error.
+        execute("insert into t (a, b) values (1, 12)");
+        assertThat(response.rowCount()).isEqualTo(1L);
+    }
```

Comment:
Similar to https://github.com/crate/crate/pull/14366#discussion_r1254112916, cannot be replaced by an `IndexerTest`